### PR TITLE
Use new context type to prevent de-opts in React

### PR DESCRIPTION
**Pull request by natew**
Using the new context type should fix performance and be forward-comparable with Concurrent mode.

### CODE REVIEW

**Comment by natew** [comment]
Updated it to also handle the createRef() issue, see: https://github.com/clauderic/react-sortable-hoc/issues/460

Only thing left would be findDOMNode, but that seems to be fine for now.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "react-sortable-hoc",
-  "version": "1.10.1",
+  "name": "@o/react-sortable-hoc",
+  "version": "1.10.4",
```
Comment (clauderic) [inline package.json]: Would you mind removing this change from this PR?

File: src/SortableContainer/index.js [hunk 2]
```diff
@@ -1029,10 +1024,16 @@ export default function sortableContainer(
...
 
+    wrappedInstance = React.createRef();
+
     render() {
-      const ref = config.withRef ? 'wrappedInstance' : null;
+      const ref = config.withRef ? this.wrappedInstance : null;
 
-      return <WrappedComponent ref={ref} {...omit(this.props, omittedProps)} />;
+      return (
+        <SortableContext.Provider value={{manager: this.manager}}>
```
Comment (clauderic) [inline src/SortableContainer/index.js]: Would you mind moving the value of the provider outside of the render method as a class property? This way it won't re-create a new object on every single render.

File: src/SortableHandle/index.js [hunk 3]
```diff
@@ -21,11 +21,13 @@ export default function sortableHandle(
...
       );
-      return this.refs.wrappedInstance;
+      return this.wrappedInstance.current;
     }
 
+    wrappedInstance = React.createRef();
+
     render() {
-      const ref = config.withRef ? 'wrappedInstance' : null;
+      const ref = config.withRef ? this.wrappedInstance : null;
```
Comment (clauderic) [inline src/SortableHandle/index.js]: In the future I think we may actually want to use `React.forwardRef()`, but for now this is an okay change

**Comment by natew** [comment]
Don't think I have time for this unfortunately, but will leave open for anyone else.

**Comment by lindskogen** [comment]
I continued this task in https://github.com/clauderic/react-sortable-hoc/pull/624 🎉

**Pull request closed without merging**